stylesheets/_component.dropzone.scss

```diff
@@ -0,0 +1,319 @@
+    transition: transform 200ms 0ms;
+}
+
+.dropzone__icon--centre {
+    left: 50%;
+    top: 0;
+    transform: translateX(-50%) translateY(25%);
+    transition: transform 200ms 50ms;
+}
+
+.dropzone__icon--right {
+    right: 0;
+    top: 0;
+    transform: rotate($dropzone-icon-rotation) translateY(75%);
+    transition: transform 200ms 100ms;
+}
+
+.dropzone__label {
+    font-family: $font-family-regular;
+    font-size: 1.25em;
+    line-height: 1;
+    margin-bottom: 0;
+    position: relative;
+}
+
+.dropzone__info {
+    font-size: .85em;
+    margin-bottom: 0;
+    position: relative;
+}
+
+.dropzone__help {
+    color: color(text, help);
+    display: inline-block;
+    font-size: .85em;
+    margin-bottom: 0;
+    margin-top: 0;
+    padding: 0 1em;
+    position: relative;
+}
+
+/**
+ * Error
+ */
+
+.dropzone__error {
+    color: color(danger);
+    font-family: $font-family-regular;
+    font-size: .85em;
+    line-height: 1.2;
+    margin: 0;
+    position: relative;
+}
+
+/**
+ * File
+ */
+
+.dropzone__file {
+    color: color(text, base);
+    max-width: 100%;
+    padding: $dropzone-file-size;
+    width: 100%;
+
+    @include respond-min($screen-small) {
+        width: 50%;
+    }
+
+    .dropzone--files-1 & {
+        margin: auto;
+        width: 100%;
+
+        @include respond-min($screen-small) {
+            width: auto;
+        }
+    }
+}
+
+.dropzone__file-inner {
+    align-items: center;
+    background-color: color(gray, off-white);
+    border-radius: $border-radius;
+    display: flex;
+    flex-grow: 1;
+    height: 100%;
+    padding: $dropzone-file-size 2em $dropzone-file-size $dropzone-file-size;
+    position: relative;
+    text-align: left;
+}
+
+.dropzone__close {
+    color: color(gray);
+    cursor: pointer;
+    padding: $dropzone-file-size;
+    position: absolute;
+    right: 0;
+    top: 0;
+
+    &:hover {
+        color: color(danger);
+    }
+}
+
+.dropzone__thumbnail {
+    border-radius: $border-radius;
+    color: color(grey);
+    height: $dropzone-thumb-size;
+    text-align: center;
+    width: $dropzone-thumb-size;
+
+    ~ .dropzone__meta {
+        width: calc(100% - #{$dropzone-thumb-size});
+    }
+}
+
+.dropzone__thumbnail--image {
+    background-size: cover;
+}
+
+.dropzone__file-icon {
+    font-size: 3em;
+    position: relative;
+    top: 50%;
+    transform: translateY(-50%);
+}
+
+.dropzone__meta {
+    padding: 0 ($dropzone-file-size * 2);
+}
+
+.dropzone__name,
+.dropzone__size,
+.dropzone__type {
+    font-size: .75em;
+    line-height: 1;
+    margin-bottom: $dropzone-file-size / 2;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
+
+.dropzone__type {
+    margin-bottom: 0;
+}
+
+.dropzone__name {
+    font-family: $font-family-regular;
+    font-size: .85em;
+    line-height: 1.1;
+    margin-bottom: $dropzone-file-size;
+}
+
+/**
+ * State
+ */
+
+@mixin dropzone-flash ($state) {
```

Comment:
It _could_, I'm not sure it should though. It needs to use the `color(...)` function which is included after we include our base mixins, so it'd throw out the consistency there, I'm also not sure it's generic enough to warrant being an "official" mixin.